### PR TITLE
Standardize geotiff / image metadata

### PR DESCRIPTION
**Pull request by craig-willis**
Changes include:
* Added netcdf, imagemagick, exiftools utilities to terrautils Docker image
* Moved create_xxx and georeferencing functions to separate modules
* Updated create_geotiff method to accept extractor_info and cleaned_md dictionaries to add standard metadata to geotiff objects (via Setmetadata)

Output of gdalinfo:
```
gdalinfo example.tif
...
Metadata:
  AREA_OR_POINT=Area
  datetime=2017-07-01T00:00:54-07:00
  experiment_name=MAC Season 4: Biomass Association Panel
  extractor_author=Solmaz Hajmohammadi <solmaz.hajmohammadi@lemnatec.de>
  extractor_description=3D to 2D Height Map conversion extractor
  extractor_name=terra.3dscanner.heightmap
  extractor_version=2.0
  sensor_id=scanner3DTop
  sensor_url=http://terraref.ncsa.illinois.edu/clowder/api/datasets/5873a7444f0cad7d81319b2b/metadata.jsonld
...

### CODE REVIEW

**Comment by craig-willis** [comment]
https://github.com/terraref/computing-pipeline/issues/268

File: terrautils/formats.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
+"""Formats
```
Comment (dlebauer) [inline terrautils/formats.py]: would be nice to have a test that takes a trivial array, exports a file in each type, and checks the properties.

File: terrautils/lemnatec.py [hunk 2]
```diff
@@ -309,23 +339,25 @@ def _standardize_gantry_system_variable_metadata(lem_md, filepath=""):
...
     properties = _standardize_with_validation("gantry_system_variable_metadata", orig, prop_map, [], filepath)  
-    
+
     # Set default scan_direction_is_positive
     if 'scan_direction_is_positive' not in properties:
         if 'position_m' in properties:
             if properties['position_m']['y'] == 0:
                 properties['scan_direction_is_positive'] = 'True'
             else:
                 properties['scan_direction_is_posfitive'] = 'False'
```
Comment (dlebauer) [inline terrautils/lemnatec.py]: Check spelling on line 349 `scan_direction_is_posfitive`

File: terrautils/spatial.py [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+
+    return [
+        (gps_bounds[2], gps_bounds[1], z_value), # upper-left
+        (gps_bounds[3], gps_bounds[1], z_value), # upper-right
+        (gps_bounds[3], gps_bounds[0], z_value), # lower-right
+        (gps_bounds[2], gps_bounds[0], z_value)  # lower-left
+    ]
+
+
+def calculate_centroid(gps_bounds):
```
Comment (dlebauer) [inline terrautils/spatial.py]: Just want to make sure that we aren't re-inventing the wheel and that we are taking advantage of existing geospatial libraries where they exist  e.g. https://shapely.readthedocs.io/en/latest/manual.html#object.centroid https://shapely.readthedocs.io/en/latest/manual.html#object.union

Not necessary for beta release, but something to keep in mind b/c the canonical implementation is more reliable and easier to maintain. 

@jterstriep could you please comment?

File: terrautils/spatial.py [hunk 4]
```diff
@@ -0,0 +1,244 @@
...
+                                        long (x) min, long (x) max )
+    """
+
+    # Get UTM information from southeast corner of field
+    SE_utm = utm.from_latlon(33.07451869, -111.97477775)
+    utm_zone = SE_utm[2]
+    utm_num  = SE_utm[3]
+
+    # TODO: Hard-coded
+    # Linear transformation coefficients
```
Comment (dlebauer) [inline terrautils/spatial.py]: Shouldn't these be in gantry fixed meta-data somewhere?

File: terrautils/lemnatec.py (hunk 2 above)
Comment (craig-willis) [reply terrautils/lemnatec.py]: Ack. Thank you.

**Comment by max-zilla** [comment]
made a note of David's requests, but with critical spelling correction am going to merge so we can deploy test things this week.

**Pull request merged**